src/hooks/useGravity.ts

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+
+import { CSSPairs, getInterpolatorForPair } from "../helpers/pairs";
+import { Gravity1DParams, gravity1D } from "../animation/gravity";
+
+type UseGravityArgs = CSSPairs & Omit<Gravity1DParams, "onUpdate">;
+
+export const useGravity = <T extends HTMLElement>({
+  from,
+  to,
+  config
+}: UseGravityArgs) => {
+  /**
+   * Store a ref to the DOM element we'll be attaching to.
+   * A user will spread this ref onto their own element, which
+   * is what allows us to directly update the style property
+   * without triggering rerenders.
+   */
+  const ref = React.useRef<T>(null);
+
+  const [stop] = React.useMemo(
```

Comment:
So @bmathews made a great point. Our components might be re-rendering a bunch, but we don't want that behavior to restart our animation. This is my first naive attempt at controlling that behavior, but I'm not convinced it'll work. If anyone has thoughts on how to preserve animation state across re-renders I'd be very interested!